src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -227,6 +251,30 @@ def get_keyring_with_caps(self, entity: AuthEntity, caps: List[str]) -> str:
             })
             if err:
                 self.mgr.log.warning(f"Unable to update caps for {entity}")
+
+            # get keyring anyway
+            ret, keyring, err = self.mgr.mon_command({
+                'prefix': 'auth get',
+                'entity': entity,
+            })
+            if err:
+                self.mgr.log.warning(f"Unable to fetch keyring for {entity}")
```

Comment:
If `auth get` fails, the keyring at the end will look like

```
[entity]
key = None
```

cause `keyring.splitlines()` will be an empty list. Did I miss anyhting?